lib/syskit/cli/log_runtime_archive.rb

```diff
@@ -81,26 +99,21 @@ def ensure_free_space(free_space_low_limit, free_space_delete_until)
                 end
             end
 
-            def process_dataset(child, full:)
-                use_existing = true
-                loop do
-                    open_archive_for(
-                        child.basename.to_s, use_existing: use_existing
-                    ) do |io|
-                        if io.tell > @max_archive_size
-                            use_existing = false
-                            break
-                        end
-
-                        dataset_complete = self.class.archive_dataset(
-                            io, child,
-                            logger: @logger, full: full,
-                            max_size: @max_archive_size
-                        )
-                        return if dataset_complete
-                    end
+            def transfering_and_archiving_candidates(candidates)
+                transfer, archive = candidates.partition do |child|
+                    self.class.archive_size_exceeded?(child)
+                end
```

Comment:
Why do you need to partition? Is there a different policy between transfer and archiving that I am not getting?